README.md

```diff
@@ -937,8 +937,10 @@ The following tables show the available configuration:
 
 | Configuration                            | Description                                                                                                                                                                         | Default            |
 |------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
-| `logbook.include`                        | Include only certain URLs (if defined)                                                                                                                                              | `[]`               |
-| `logbook.exclude`                        | Exclude certain URLs (overrides `logbook.include`)                                                                                                                                  | `[]`               |
+| `logbook.predicate.include.urls`         | Include only certain URLs (if defined)                                                                                                                                              | `[]`               |

```

Comment:
Is my understanding correct, that URL- and method-matching can't be combined for a single match, i.e. it's not possible with this configuration to e.g. say `include POST requests to /api/v1`. Is that correct?

Reply:
On the contrary, that's how they'll work. If someone configures both `url` and `method` in `logbook.predicate.include` than only the combination of the two will be logged. See examples in [IncludeTest](https://github.com/zalando/logbook/pull/1606/files#diff-6b8212a17c483142caa6609499188dc17270b6bbd4f76a6fd01e811630272c56R66-R69)

Reply:
Moving the discussion back here :smile: 

> Good, but that means I can configure something like: only log "POST to
/api/v1" and "GET /search", right?


You're right here, the predicates are not grouped by routes with this implementation. All `include` predicates will be merged together (method and urls ones separately first). So if someone configures
`logbook.predicate.include.urls=/api/v1,/api/v2` and `logbook.predicate.include.methods=GET,POST`
Then it'll allow all combinations of paths and methods. 

I do like the idea of grouping the predicates per route, where per one path, several methods can be added. Then it'll work as you mentioned.

Reply:
I was thinking more along the lines of this:

```yaml
logbook.predicate.include:
  - path: /api/v1
    method: POST
  - path: /search
    method: GET
```

The url/path wouldn't be special here, in fact it could even be optional (same as method):

```yaml
logbook.predicate.include:
  - method: DELETE # all paths
  - path: /api/v1 # all methods
```

Reply:
Agreed, this would be quite flexible. I'll add it this way :+1:  (after my vacation).

Reply:
@whiskeysierra I've updated the PR to work as we discussed here